experimental/MatroidRealizationSpaces/src/SelfProjectingMatroids.jl

```diff
@@ -0,0 +1,336 @@
+    defining_ideal(RS::MatroidRealizationSpaceSelfProjecting)
+
+The ideal of the matroid realization space `RS`.
+"""
+defining_ideal(RS::MatroidRealizationSpaceSelfProjecting) = RS.defining_ideal
+
+@doc raw"""
+    inequations(RS::MatroidRealizationSpaceSelfProjecting)
+
+Generators of the localizing semigroup of `RS`.
+These are the polynomials that need to be nonzero in any realization.
+"""
+inequations(RS::MatroidRealizationSpaceSelfProjecting) = RS.inequations
+
+@doc raw"""
+    ambient_ring(RS::MatroidRealizationSpaceSelfProjecting)
+
+The polynomial ring containing the ideal `defining_ideal_sp(RS)` and the polynomials in `inequations(RS)`.
+"""
+ambient_ring(RS::MatroidRealizationSpaceSelfProjecting) = RS.ambient_ring
+
+
+
+@doc raw"""
+    selfproj_realization_matrix(RS::MatroidRealizationSpaceSelfProjecting)
+
+A matrix with entries in ambient_ring_sp(RS) whose columns, when filled in with values satisfying equalities
+from `defining_ideal(RS)` and inequations from `inequations(RS)`, form a self-projecting realization for the matroid.
+"""
+selfproj_realization_matrix(RS::MatroidRealizationSpaceSelfProjecting) = RS.selfproj_realization_matrix
+
+function satisfies_disjointbasisproperty(mat::Matroid)::Bool
+    dmat = dual_matroid(mat)
+    boo = false
+    for b in bases(dmat)
+        boo = false
+        for e in bases(mat)
+            if issubset(e,b)
+                boo = true
+                break
+            end
+        end
+        if !boo 
+            break
+        end
+    end
+    return boo;
+end
+
+#A function to test if a matroid is self-projecting
+function is_selfprojecting(mat::Matroid)::Bool
+    k = rank(mat)
+    n = length(matroid_groundset(mat))
+    boo_basis = false
+    for b in bases(mat)
+        boo_basis = false
+        for c in bases(mat)
+            if intersect(b,c) == []
+                boo_basis = true
+                break
+            end
+        end
+        if !boo_basis 
+            break
+        end
+    end
+    boo_flats = true
+    for f in flats(mat,k-1)
+        boo_flats = true
+        for g in flats(mat,k-1)
+            if length(union(f,g)) == n-1
+                boo_flats = false
+                break
+            end
+        end
+        if !boo_flats 
+            break
+        end
+    end
+    return boo_basis && boo_flats;
+end
+@doc raw"""
+    is_selfprojecting(mat::Matroid)
+
+Returns a boolean which states whether the given matroid satisfies the property to be self-projecting.
+# Examples
+```jldoctest
+julia> M = fano_matroid();
+```
+"""
+
+#this function is not properly tested, since it did not terminate for intersting examples.
+function selfproj_realization_ideal(m::Matroid;saturate::Bool = false)::Ideal
+    @warn "This function is slow except for small matroids!"
+    if !is_selfprojecting(m) 
+      error("The given matroid is not self-projecting.") #Is it too costly to have this check?! How about an option to check, which can be turned off?
```

Comment:
it's quite common to have a kwarg `check::Bool=true` when the check can take a while

Reply:
I don't quite understand what you mean: Do you mean I should add another optional argument ``check::Bool=true`` with
``if check  && ! is_selfprojecting(m)
error("The given matroid is not selfprojecting.")
end``
 ?

Reply:
yes exactly. 
But related to my other comments, instead of throwing an error use the 
@req macro
```suggestion
  check && @req is_self_projecting(m) "The given matroid is not self projecting"
```